modules/genflow-api/src/main/java/com/reprezen/genflow/api/template/GenTemplate.java

```diff
@@ -1,928 +1,933 @@
+
+import org.apache.commons.io.FilenameUtils;
+
+import com.google.common.base.Charsets;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.io.Files;
+import com.reprezen.genflow.api.GenerationException;
+import com.reprezen.genflow.api.outputitem.IOutputItem;
+import com.reprezen.genflow.api.source.ILocator;
+import com.reprezen.genflow.api.source.ISource;
+import com.reprezen.genflow.api.target.GenTarget;
+import com.reprezen.genflow.api.target.GenTargetNamedSource;
+import com.reprezen.genflow.api.target.GenTargetPrerequisite;
+import com.reprezen.genflow.api.template.builders.DynamicGeneratorBuilder;
+import com.reprezen.genflow.api.template.builders.DynamicGeneratorBuilder.DynamicGeneratorSpec;
+import com.reprezen.genflow.api.template.builders.NamedSourceBuilder;
+import com.reprezen.genflow.api.template.builders.NamedSourceBuilder.NamedSourceSpec;
+import com.reprezen.genflow.api.template.builders.OutputItemBuilder;
+import com.reprezen.genflow.api.template.builders.OutputItemBuilder.OutputItemSpec;
+import com.reprezen.genflow.api.template.builders.ParameterBuilder;
+import com.reprezen.genflow.api.template.builders.ParameterBuilder.ParameterSpec;
+import com.reprezen.genflow.api.template.builders.PrerequisiteBuilder;
+import com.reprezen.genflow.api.template.builders.PrerequisiteBuilder.PrerequisiteSpec;
+import com.reprezen.genflow.api.template.builders.PrimarySourceBuilder;
+import com.reprezen.genflow.api.template.builders.PrimarySourceBuilder.PrimarySourceSpec;
+import com.reprezen.genflow.api.template.builders.PropertyBuilder;
+import com.reprezen.genflow.api.template.builders.StaticResourceBuilder;
+import com.reprezen.genflow.api.template.builders.StaticResourceBuilder.StaticResourceSpec;
+import com.reprezen.genflow.api.template.config.GenTargetConfigUtil;
+import com.reprezen.genflow.api.template.config.GenTemplateConfig;
+import com.reprezen.genflow.api.template.config.NamedSourceConfig;
+import com.reprezen.genflow.api.template.config.OutputItemConfig;
+import com.reprezen.genflow.api.template.config.ParameterConfig;
+import com.reprezen.genflow.api.template.config.PrerequisiteConfig;
+import com.reprezen.genflow.api.template.config.PrimarySourceConfig;
+import com.reprezen.genflow.api.template.config.StaticResourceConfig;
+import com.reprezen.genflow.api.trace.GenTemplateTrace;
+import com.reprezen.genflow.api.trace.GenTemplateTraceBuilder;
+import com.reprezen.genflow.api.trace.GenTemplateTraces;
+import com.reprezen.genflow.api.util.Eval;
+import com.reprezen.genflow.api.util.Eval.VarName;
+import com.reprezen.genflow.api.util.FileUtils;
+import com.reprezen.genflow.api.util.Strings;
+import com.reprezen.genflow.api.util.TypeUtils;
+
+public abstract class GenTemplate<PrimaryType> extends AbstractGenTemplate {
+
+	public static final String OUTPUT_FILES_PARAM = "outputFileNames";
+	public static final String IMPORT_MAPPINGS = "importMappings";

```

Comment:
@neerajbhusare , FYI, GitHub shows that this was a "large diff", too large to load by default until I viewed without whitespace changes.

Did you accidentally change the formatting of this file?

Reply:
I am using the standard Eclipse formatter as suggested in the setup doc - https://modelsolv.atlassian.net/wiki/spaces/modsquad/pages/18317322/Workbench+Setup

The formatter works strange for this particular file, it is ok for all the other files I change. Perhaps this could be because the file was formatted with a custom formatter before - the last commit was by @andylowry (ZEN-4490 fully-automated source formatting).

Anyways, I have fixed it for now. You should only see the changes I made.

Reply:
@tedepstein It still shows a whole lot of changes. Not sure how to fix it. @ghillairet Any clue?